frontend/src/components/common/FormikPersist.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { FormikProps, connect } from 'formik';
+import debounce from 'lodash/debounce';
+import isEqual from 'react-fast-compare';
+import { AES, enc } from 'crypto-js';
+
+export interface PersistProps {
+  name: string;
+  secret: string;
+  initialValues: any;
+  debounce?: number;
+  isSessionStorage?: boolean;
+  writeOnly?: boolean;
+}
+
+class PersistImpl extends React.Component<PersistProps & { formik: FormikProps<any> }, {}> {
+  static defaultProps = {
+    debounce: 300,
+  };
+
+  //Encrypt the formdata before saving.
+  //TODO: in the event the security of this data becomes a concern, this data should be persisted server-side.
+  saveForm = debounce((data: FormikProps<{}>) => {
+    if (this.props.isSessionStorage) {
+      window.sessionStorage.setItem(
+        this.props.name,
+        AES.encrypt(JSON.stringify(data), this.props.secret).toString(),
```

Comment:
This level of security obtainable here is entirely based on the strength of the key. For now I'm just using the JWT subject to encode this data. We could replace this with something more secure in the future if the security of this formik data becomes a concern.

Also, it would be easy to wipe all saved data in the event that the isSensitive field is ticked, which would be the easiest option.

Reply:
This is probably overkill as all this is client-side, but I love it anyway lol.  Cool feature.